homeassistant/components/knx/number.py

```diff
@@ -0,0 +1,122 @@
+"""Support for KNX/IP numeric values."""
+from __future__ import annotations
+
+import voluptuous as vol
+from xknx import XKNX
+from xknx.devices import NumericValue
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.const import CONF_NAME, CONF_TYPE, STATE_UNAVAILABLE, STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import CONF_RESPOND_TO_READ, CONF_STATE_ADDRESS, DOMAIN, KNX_ADDRESS
+from .knx_entity import KnxEntity
+from .schema import NumberSchema
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up number entities for KNX platform."""
+    if not discovery_info or not discovery_info["platform_config"]:
+        return
+
+    platform_config = discovery_info["platform_config"]
+    xknx: XKNX = hass.data[DOMAIN].xknx
+
+    entities = []
+    for entity_config in platform_config:
+        entities.append(KNXNumber(xknx, entity_config))
+
+    async_add_entities(entities)
+
+
+def _create_expose_sensor(xknx: XKNX, config: ConfigType) -> NumericValue:
+    """Return a KNX NumericValue to be used within XKNX."""
+    return NumericValue(
+        xknx,
+        name=config[CONF_NAME],
+        group_address=config[KNX_ADDRESS],
+        group_address_state=config.get(CONF_STATE_ADDRESS),
+        respond_to_read=config[CONF_RESPOND_TO_READ],
+        value_type=config[CONF_TYPE],
+    )
+
+
+class KNXNumber(KnxEntity, NumberEntity, RestoreEntity):
+    """Representation of a KNX number."""
+
+    def __init__(self, xknx: XKNX, config: ConfigType) -> None:
+        """Initialize a KNX number."""
+        self._device: NumericValue
+        super().__init__(_create_expose_sensor(xknx, config))
+        self._unique_id = f"{self._device.sensor_value.group_address}"
+
+        self._min_value: float = max(
+            config.get(
+                NumberSchema.CONF_MIN,
+                self._device.sensor_value.dpt_class.value_min,
+            ),
+            self._device.sensor_value.dpt_class.value_min,
+        )
+        self._max_value: float = min(
+            config.get(
+                NumberSchema.CONF_MAX,
+                self._device.sensor_value.dpt_class.value_max,
+            ),
+            self._device.sensor_value.dpt_class.value_max,
+        )
```

Comment:
`self._device.sensor_value.dpt_class.value_max` can (for 4byte float DPT 14) be `float('inf')` (value_min -> `float('-inf')`). This throws an error on startup
> [homeassistant.components.websocket_api.messages] Unable to serialize to JSON. Bad data found at $.result[18](State: number.test_number).attributes.min=-inf(<class 'float'>, $.result[18](State: number.test_number).attributes.max=inf(<class 'float'>

I'm not sure how to handle this properly. Any suggestions?

Reply:
I'd just turn inf into none.

Reply:
Return value of the property is defined `float`, not `Optional[float]`.

Reply:
I just tried this:
Returning `None` renders a number card that only accepts `0` as valid value.

Reply:
Seem bad.

Reply:
Avoided this by Schema validation. Values supporting infinity now require a user set min/max.